oclif/plugin-plugins
# fix: use internal npm for pkg checks

When installing a plugin using an "unfriendly" plugin name, `plugins install` needs to call npm to verify that the pkg exists in the registry. Until now it's been doing that by executing `npm show <pkg-name> dist-tags`, which works fine unless you don't have `npm` installed globally on your system.

This PR updates this check to use the `npm` version pinned in and shipped with `plugin-plugins`, this way the user doesn't need to install npm and also helps avoid weird issues if a user has an old npm version or a new breaking change is released.

## Testing notes:
1. Checkout this PR locally
2. Link plugin (`plugins link .`)
3. Try installing a plugin using an "unfriendly" name and verify the debug output to see which node/npm is being used.

**Suggestion**
set `DEBUG='@oclif/plugins'` when testing to see additional debug output.
![Screenshot 2023-04-27 at 11 22 37](https://user-images.githubusercontent.com/6853656/234891863-be4c50c5-16e1-4530-9552-c7e4bb792374.png)


## Possible scenarios
1) Install CLI via mac/win installer or tarballs
Installers/tarballs include a node executable and `plugins install <unfriendly-name` will perform the call to the pinned npm version in `plugin-plugins/package.json` using the include node binary.

2) Install CLI via npm (so user has node installed)
`plugins install <unfriendly-name>` will use the global node binary to call the pinned npm version in `plugin-plugins/package.json`


It doesn't matter if the user has `npm` installed globally, `plugin-plugins` should always use the pinned version included by the plugin.

## What's an "unfriendly" pkg name?
If you define an npm pkg scope in your CLI (`pjson.oclif.scope`), you can use a shorter name when installing plugins published under that scope.
https://github.com/oclif/plugin-plugins/blob/main/src/plugins.ts#L258

**Example**:
The Salesforce CLI defines `salesforce` as an npm scope here:
https://github.com/salesforcecli/sfdx-cli/blob/main/package.json#L27

To install `@salesforce/plugin-data` you can run:
`sfdx plugins install data` and plugin-plugins resolves the plugin name.

@W-13018874@
